Fix deleting of zipped Dags in Serialized Dag Table

ephraimbuddy opened the pull request:
The file locations of DAGs in zipped folders are not correctly listed when removing deleted dags from serialized dag table thus the delete query for deleting deleted dags from serialized DAGs is deleting dags in zipped folders. Likewise DagCode.remove_deleted_code

Related: https://github.com/apache/airflow/issues/18114#issuecomment-918090551



---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

jedcunningham commented on airflow/models/dag.py:
```diff
@@ -2814,10 +2815,24 @@ def deactivate_deleted_dags(cls, alive_dag_filelocs: List[str], session=None):
+        for fileloc in alive_dag_filelocs:
+            if zipfile.is_zipfile(fileloc):
+                with zipfile.ZipFile(fileloc) as z:
+                    dag_filelocs.extend(
```
I wonder if this should be done once in `_refresh_dag_dir` instead of done twice here and in `remove_deleted_dags`?

ephraimbuddy replied:
Mistake adding it here, it's the `DagCode.remove_deleted_dags` that needed it too. Will move the logic to `_refresh_dag_dir`. Thanks

ephraimbuddy committed:
The file locations of DAGs in zipped folders are not correctly listed when removing deleted dags from
serialized dag table thus the delete query for deleting deleted dags from serialized DAGs is deleting
dags in zipped folders. Likewise DagCode.remove_deleted_code

This PR fixes it by listing all the file paths as stored in SDM so that the delete query will work properly

fixup! The file locations of DAGs in zipped folders are not correctly listed when removing deleted dags from serialized dag table thus the delete query for deleting deleted dags from serialized DAGs is deleting dags in zipped folders. Likewise DagCode.remove_deleted_code

jedcunningham commented on airflow/dag_processing/manager.py:
```diff
@@ -660,13 +661,29 @@ def _refresh_dag_dir(self):
                 self.clear_nonexistent_import_errors()
             except Exception:
                 self.log.exception("Error removing old import errors")
+            # Check if file path is a zipfile and get the full path of the python file.
```
```suggestion

            # Check if file path is a zipfile and get the full path of the python file.
```
nit

ephraimbuddy committed:
Update airflow/dag_processing/manager.py

Co-authored-by: Jed Cunningham <66968678+jedcunningham@users.noreply.github.com>
